src/components/FolioNumber/FolioNumber.vue

```diff
@@ -0,0 +1,264 @@
+            <v-btn
+              large color="primary"
+              class="save-button"
+              type="submit"
+              value="Submit"
+            >
+              <span>Save</span>
+            </v-btn>
+            <v-btn
+              large outlined color="primary"
+              class="cancel-button"
+              @click="onCancelClicked()"
+            >
+              <span>Cancel</span>
+            </v-btn>
+          </div>
+        </v-col>
+      </v-row>
+    </v-form>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Emit, Prop, Vue, Watch } from 'vue-property-decorator'
+
+@Component({})
+export default class FolioNumber extends Vue {
+  // Props
+  @Prop({ default: null }) readonly initialValue: string
+  @Prop({ default: null }) readonly originalValue: string
+  @Prop({ default: false }) readonly hideActions: boolean
+  @Prop() readonly editLabel: string
+  @Prop() readonly editedLabel!: string
+  @Prop({ default: false }) readonly invalidSection!: boolean
+
+  // Local properties
+  private isEditing = false
+  private formValid = false
+  private folioNumber = ''
+  private dropdown = false
+
+  // Validation rules
+  private readonly rules: Array<Function> = [
+    (v: string) => (!v || v.length <= 30) || 'Maximum 30 characters reached'
+  ]
+
+  /** Whether folio number has changed from original value. */
+  private get hasFolioNumberChanged (): boolean {
+    const fn = this.folioNumber || null
+    const ov = this.originalValue || null
+    return (fn !== ov)
+  }
+
+  /** When Undo is clicked, restores the folio number to the original value. */
+  private onUndoClicked (): void {
+    this.folioNumber = this.originalValue
+    this.emitNewFolioNumber()
+    this.emitHaveChanges()
+    this.isEditing = false
+  }
+
+  /** When form is submitted, accepts the entered folio number. */
+  private onFormSubmit (): void {
+    if (this.formValid) {
+      this.emitNewFolioNumber()
+      this.emitHaveChanges()
+      this.isEditing = false
+    }
+  }
+
+  /** When Cancel is clicked, undoes the current change. */
+  private onCancelClicked (): void {
+    this.folioNumber = this.initialValue
+    this.isEditing = false
+  }
+
+  /** Updates local value initially and when prop has changed. */
+  @Watch('initialValue', { deep: true, immediate: true })
+  private onInitialValueChanged (): void {
+    this.folioNumber = this.initialValue
+  }
+
+  /** Informs parent of new folio number. */
+  @Emit('newFolioNumber')
+  private emitNewFolioNumber (): string {
+    return this.folioNumber
+  }
+
+  /** Informs parent whether we have changes. */
+  @Emit('haveChanges')
+  private emitHaveChanges (): boolean {
+    return this.hasFolioNumberChanged
+  }
+
+  /** Informs parent whether we are in editing mode. */
+  @Watch('isEditing')
+  @Emit('isEditing')
+  private emitIsEditing (val: boolean): void {}
+}
+</script>
+
+<style lang="scss" scoped>
+@import '@/assets/styles/theme.scss';
```

Comment:
No action required here!

Just sharing a thought, in the Name Request UI when we import these components it throws errors because this is pulling the `assets` from root. It works good in create/edit where the assets file structure is the same but in Name Request, one of the sub directories is different. 

That being said, down the road we will need to figure this one out, maybe using an external style sheet that needs to be a dependency?? i'm not sure 🤷